src/Symfony/Component/Lock/Store/RetryTillSaveStore.php

```diff
@@ -37,8 +42,12 @@ class RetryTillSaveStore implements StoreInterface, LoggerAwareInterface
      * @param int            $retrySleep Duration in ms between 2 retry
      * @param int            $retryCount Maximum amount of retry
      */
-    public function __construct(StoreInterface $decorated, int $retrySleep = 100, int $retryCount = PHP_INT_MAX)
+    public function __construct($decorated, int $retrySleep = 100, int $retryCount = PHP_INT_MAX)
```

Comment:
You could type-hint for `PersistStoreInterface` and remove the type-check below. Implementations of the old `StoreInterface` do inherit `PersistStoreInterface` and thus also fulfill the requirement.

Reply:
You are right.